lib/structs/responses/common.cpp

```diff
@@ -436,6 +436,10 @@ parse_timeline_events(const json &events,
                     break;
                 }
                 case MsgType::Unknown: {
+                    container.emplace_back(events::RoomEvent<events::msg::Unknown>(e));
+                    break;
+                }
+                case MsgType::Invalid: {
                     container.emplace_back(events::RoomEvent<events::Unknown>(e));
```

Comment:
Hm, so I think having an invalid state for the msgtype is good, but I think we should just drop such events. There is little reason to store invalid events in the database and these are clear spec violations that have no valid representation. What do you think?

Reply:
Sure, I was just trying to preserve current behavior which was somewhat preserving invalid events.

Reply:
I think preserving unknown events and unparseable non-msg events is enough. But thanks!